pkg/validator/validators.go

```diff
@@ -266,3 +268,82 @@ func (kc *policyValidator) upstreamTrafficSettingValidator(req *admissionv1.Admi
 
 	return nil, nil
 }
+
+func meshRootCertificateValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	switch req.Operation {
+	case admissionv1.Create:
+		newMRC := &configv1alpha2.MeshRootCertificate{}
+		if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(newMRC); err != nil {
+			return nil, err
+		}
+
+		err := validateMRCOnCreate(newMRC)
+		return nil, err
+	case admissionv1.Update:
+		newMRC, oldMRC := &configv1alpha2.MeshRootCertificate{}, &configv1alpha2.MeshRootCertificate{}
+		if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(newMRC); err != nil {
+			return nil, err
+		}
+		if err := json.NewDecoder(bytes.NewBuffer(req.OldObject.Raw)).Decode(oldMRC); err != nil {
+			return nil, err
+		}
+
+		err := validateMRCOnUpdate(oldMRC, newMRC)
+		return nil, err
+	}
+	return nil, nil
+}
+
+func validateMRCOnCreate(mrc *configv1alpha2.MeshRootCertificate) error {
```

Comment:
Please add a field to prevent creating an `active` MRC if one already exists

Reply:
I was planning on addressing more validation in a follow-up PR. When I created this PR I was blocked on https://github.com/openservicemesh/osm/pull/5127. Since I am no longer blocked on this change, I can include the validation that required the MRC methods in the k8s client in this PR.

Reply:
Ah, if you're planning to add it in a followup PR, that's fine; just add a TODO comment and I'll approve

Reply:
@keithmattix I went ahead and added the active MRC check